sched/signal/sig_dispatch.c

```diff
@@ -319,7 +320,8 @@ int nxsig_tcbdispatch(FAR struct tcb_s *stcb, siginfo_t *info)
 
   /************************** MASKED SIGNAL ACTIONS *************************/
 
-  masked = (bool)sigismember(&stcb->sigprocmask, info->si_signo);
+  masked = (nxsig_ismember(&stcb->sigprocmask, info->si_signo) <= 0) ?
+           false : true;
```

Comment:
There is really no simple solution to make this "perfect" without adding more error handling logic.  If an error occurs, we will say that the signal is not masked.